### DISCUSSION TITLE
Mongo DB execute search using the text index.

### DISCUSSION THREAD

**Discussion by Lishenga** [mikro-orm/mikro-orm#2476] [Q&A] [answered]
Hi, I love the job you're doing with Mikro ORM it is an awesome ORM. Just a quick question, does MikroORM support searches using the text index in MongoDB. Like I can do this using MongoSH 
```ts
db.item.find( { $text: { $search: "OVES Connected eIoT" } } )
```

Is it possible to execute the above query using MikroORM and if it is possible could you provide me an example of such a query using MikroORM.

Thank you in advance.

**Comment by B4nan** [comment, marked as answer]
You could do any query directly via driver, and map the entities via `em.map()`:

```ts
// you can also just do `em.getCollection()` if you have the EM correctly typed
const res = em.getDriver().getCollection(User.name).find({ $text: { $search: "OVES Connected eIoT" } });
const entities = res.map(row => em.map(User, row));
```

**Reply by Lishenga** [reply]
Am getting this error when I try to use your example above
<img width="925" alt="Screenshot 2021-11-29 at 11 48 51" src="https://user-images.githubusercontent.com/33782195/143836356-6b8feba6-d815-4533-8c26-a2864e58b7ac.png">

What could I be doing wrong ?

**Reply by B4nan** [reply]
Ah I see, just use `em.getCollection()` then, it is actually on the connection level, not driver, so you would have to do `em.getConnection().getCollection()`.

**Reply by B4nan** [reply]
And regular queries via EM work fine? I'd expect those to fail the same way if your connection requires auth and you don't provide it.
